### PR TITLE
Fix lookup symbol for enum members to search a correct code definition

### LINKED ISSUE #94284: Ctrl/Cmd + left-clicking on user-defined enum value takes you to docs instead of code definition
### Tested versions

Reproducible in 4.3.beta3

### System information

Godot v4.3.beta3 - Windows 10.0.22631 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1660 SUPER (NVIDIA; 32.0.15.5599) - Intel(R) Core(TM) i7-10700F CPU @ 2.90GHz (16 Threads)

### Issue description

When you ctrl/cmd + left-click on a reference to a user-defined enum value in the Script Editor, it opens up a new tab with the documentation for the class that contains the user-defined enum, instead of taking you to the place in the code where that enum value is defined.

Furthermore, if the containing script does not have a class name, it opens up a blank documentation tab which can't be closed with the hotkey (but can still be closed via the right-click context menu).

enum_docs

While it is super cool that you can view a doc page for your own classes (I have no idea how else to trigger this behaviour but it's really cool!), this behaviour is inconsistent with the regular behaviour of ctrl/cmd + left-clicking on user-written types/values, which all take you to the place in the code that defines the item.

### Steps to reproduce

- Open up the attached project, and open up `test_script.gd` in the Script Editor.
- Ctrl/Cmd + left-click on `MyEnum.HELLO` on line 6.
- Notice that a new documentation tab opens up.
- Erase the class name in the script, save, and repeat step 2.
- Notice that a new _blank_ documentation tab is created, and cannot be closed with the hotkey.

### Minimal reproduction project (MRP)

[bug-report.zip](https://github.com/user-attachments/files/16199173/bug-report.zip)

### PR DESCRIPTION
**Pull request by Chaosus**
- Fix https://github.com/godotengine/godot/issues/94284

enum_lookup

### CODE REVIEW

**Comment by dalexeev** [comment]
I guess this isn't critical for the 4.3 release? We could move this to 4.4 and cherry-pick it for 4.3.x later?

**Review by HolonProduction — CHANGES_REQUESTED** [review]
This breaks lookup for builtin enums. It isn't directly obvious, since we try to lookup the symbol as class constant before doing anything with the base (this seems questionable and will probably lead to some edge case bugs, but that's a different issue).

Here is a minimal example script:
```gdscript
extends Node

func _init():
    Control.SizeFlags.SIZE_EXPAND
```

With this PR looking up SIZE_EXPAND will fail. Solution should be falling back to the old behaviour if the class type ~or script path~ isn't set.

**Comment by Chaosus** [comment]
@HolonProduction Fixed

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -128,14 +128,16 @@ static GDScriptParser::DataType make_script_meta_type(const Ref<Script> &p_scrip
...
-static GDScriptParser::DataType make_enum_type(const StringName &p_enum_name, const String &p_base_name, const bool p_meta = false) {
+static GDScriptParser::DataType make_enum_type(const StringName &p_enum_name, const String &p_base_name, const bool p_meta = false, GDScriptParser::ClassNode *p_class = nullptr, const String &p_script_path = "") {
 	GDScriptParser::DataType type;
 	type.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
 	type.kind = GDScriptParser::DataType::ENUM;
 	type.builtin_type = p_meta ? Variant::DICTIONARY : Variant::INT;
 	type.enum_type = p_enum_name;
 	type.is_constant = true;
 	type.is_meta_type = p_meta;
+	type.class_type = p_class;
```
Comment (HolonProduction) [inline modules/gdscript/gdscript_analyzer.cpp:139]: Given that we have make_global_enum type and make_native_enum_type which use this method as a base, I'd rather put adding a class into a new method make_class_enum_type and keep this method as the "common" logic

Comment (Chaosus) [reply modules/gdscript/gdscript_analyzer.cpp:139]: Okay.. done, Check again

**Review by HolonProduction — APPROVED** [review]
- tested all edge cases that I could think of
- code content makes sense
- not well versed enough in C++ to give a thorough review on code quality but since it's in line with other stuff in this file it should be fine

Overall LGTM

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**